File: src/main/java/com/serfagab/entities/Usuario.java
package com.serfagab.entities;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "tbl_usuario")
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    private Integer idUsuario;

    @Column(name = "nombres")
    private String nombres;

    @Column(name = "apellidos")
    private String apellidos;

    @Column(name = "login")
    private String login;

    @Column(name = "clave")
    private String clave;

    @Column(name = "email")
    private String email;

    @Column(name = "activo")
    private Boolean activo = true;

    @ManyToOne
    @JoinColumn(name = "id_tipo")
    private Tipo tipo;
}

File: frontend-angular/src/app/models/usuario.ts
export interface Tipo {
  idTipo: number;
  descripcion: string;
}

export interface Usuario {
  idUsuario: number;
  nombres: string;
  apellidos: string;
  login: string;
  clave: string;
  email: string;
  activo: boolean;
  tipo: Tipo;
}